### ISSUE TITLE
Use native vacuum cleaner controls in iOS 18

### ISSUE THREAD

**Issue by azmke** [merdok/homebridge-miot#621]
Robot vacuum cleaners will be natively supported by the Home app in iOS 18, so they can participate in automations and scenes, and can be activated by a user’s voice using Siri.

Source: https://www.apple.com/newsroom/2024/06/updates-to-the-home-experience-elevate-entertainment-and-bring-more-convenience/

I think it would be cool to update the plugin to be able to use the new controls 😊

**Comment by merdok** [comment]
Thanks for the info! That sounds really good, will certainly have a look at this as soon as homebridge will support those devices. Or maybe it already does?

**Comment by frtzmsr** [comment]
Really appreciate your work!
Would love if you could add the iOS 18 support for vacuums!

**Comment by merdok** [comment]
Does already homebridge support the vacuums?

**Comment by itsJumaah** [comment]
Yes it does

**Comment by merdok** [comment]
Nice, would you have a link which I check out?

**Comment by azmke** [comment]
Hi @itsJumaah,

thank you for taking the time to reply. Can you tell me where you got the information from?

I couldn't find any indication of support for vacuum robots in neither the code of the [Homebridge](https://github.com/homebridge/homebridge) nor the [HAP-NodeJS](https://github.com/homebridge/HAP-NodeJS) repository.

Even on the [official Apple page for iOS 18](https://www.apple.com/ios/ios-18/) it still says that "robot vacuum cleaner support" is "coming later this year".

I would be very happy if you could help me. Preferably with a concrete code snippet. Can't wait to integrate my robot vacuum cleaner “properly” into the Home app.

**Comment by Semir333** [comment]
+1

**Comment by merdok** [comment]
As far as I am aware the native support for vacuum cleaners is still missing in HomeKit

**Comment by MrNoooooooob** [comment]
With iOS 18.3, the vacuum cleaners will be unlocked in 2025 and then only via Matter. Homebridge should slowly change course as Apple moves away from its architecture.

**Comment by merdok** [comment]
That is odd, we will have to wait and see in what direction this will go... I have the feeling that in general Apple seems to be going away from HomeKit...

**Comment by MrNoooooooob** [comment]
> That is odd, we will have to wait and see in what direction this will go... I have the feeling that in general Apple seems to be going away from HomeKit...

I'm not sure, but Apple even announced in a press release that they are moving away from their architecture.

It's logical that Apple is ditching its framework, as Matter is the better alternative, and I hope that Homebridge is working on a Matter system, as something similar already exists.

**Comment by merdok** [comment]
Yeah it would make sense to provide matter support in homebridge, otherwise I am not sure how the future will be.

**Comment by MrNoooooooob** [comment]
Unfortunately, we have to wait and see what happens next.

**Comment by Kief5555** [comment]
@merdok Apple added HomeKit support for matter robot vacuums.

**Comment by merdok** [comment]
@Kief5555 the issue is that it first needs to be supported by homebridge. And the issue is, as already it was discussed here, that homebridge would need to switch to matter, as the robot vacuums in HomeKit work only using matter.

**Comment by Kief5555** [comment]
> [@Kief5555](https://github.com/Kief5555) the issue is that it first needs to be supported by homebridge. And the issue is, as already it was discussed here, that homebridge would need to switch to matter, as the robot vacuums in HomeKit work only using matter.

Ah shoot! Sorry I thought it was implemented already, just read https://github.com/homebridge/homebridge/issues/3228. My mistakes.

**Comment by azmke** [comment]
I recommand reading the discussion in homebridge/HAP-NodeJS#1073. Robot vacuum cleaner support is not coming to HomeKit nor Homebridge anytime soon, sadly.

**Comment by Kief5555** [comment]
With the release of HB v2, I believe this should now be possible? @merdok 😄

**Comment by merdok** [comment]
Interesting, as I did not dig deeper into it yet, anyone knows if robot vacuums are already supported or will they come in a later update?

**Comment by Kief5555** [comment]
> Interesting, as I did not dig deeper into it yet, anyone knows if robot vacuums are already supported or will they come in a later update?

The documentations appears to be available [here](https://github.com/homebridge-plugins/homebridge-matter/wiki/Section-12-Robotic)

**Comment by merdok** [comment]
Very nice! Will check it out. Thanks!

**Comment by Kief5555** [comment]
@merdok Curious if you're open to accepting PR's that has (some/most) code being generated by AI.  Totally understand that people (like myself ironically) dislike fully based AI-generated PRs.

**Comment by merdok** [comment]
As long as the code works and is tested then it is fine😉

**Comment by Kief5555** [comment]
<img width="1179" height="2556" alt="Image" src="https://github.com/user-attachments/assets/9f0fca13-8814-48fb-8203-120a66c209fd" />

Looks promising so far
